splunk_metric/test_splunk_metric.py

```diff
@@ -1578,3 +1578,148 @@ def _mocked_auth_session_to_check_instance_config(instance):
         })
 
         self.assertEqual(len(self.metrics), 2)
+
+
+class TestSplunkConfigMaxQueryChunkSecHistory(AgentCheckTest):
+    """
+    Splunk metric check should use the max query chunk sec in past mode
+    """
+    CHECK_NAME = 'splunk_metric'
+
+    def test_checks(self):
+        self.maxDiff = None
+
+        config = {
+            'init_config': {
+                'default_restart_history_time_seconds': 3600,
+                'default_max_query_chunk_seconds': 300,
+                'default_initial_history_time_seconds': 86400,
+                'unique_key_fields': None
+            },
+            'instances': [
+                {
+                    'url': 'http://localhost:13001',
+                    'username': "admin",
+                    'password': "admin",
+                    'saved_searches': [{
+                        "name": "metrics",
+                        "parameters": {},
+                        'max_restart_history_seconds': 3600,
+                        'max_query_chunk_seconds': 300,
+                        'initial_history_time_seconds': 86400,
+                        'unique_key_fields': None
+                    }],
+                    'tags': ["checktag:checktagvalue"]
+                }
+            ]
+        }
+
+        # Used to validate which searches have been executed
+        test_data = {
+            "time": 0,
+            "earliest_time": ""
+        }
+
+        def _mocked_current_time_seconds():
+            return test_data["time"]
+
+        def _mocked_dispatch_saved_search_dispatch(*args, **kwargs):
+            earliest_time = args[5]['dispatch.earliest_time']
+            if test_data["earliest_time"] != "":
+                self.assertEquals(earliest_time, test_data["earliest_time"])
+            if test_data["earliest_time"] == "2017-03-08T00:00:00.000000+0000":
```

Comment:
Is this the initial run? Perhaps just add a comment here that this checks for the initial run and store "2017-03-08T00:00:00.000000+0000" in a constant called something like `INITIAL_EARLIEST_TIME`